transport/dgit/client.go

```diff
@@ -61,18 +64,20 @@ func NewLocalClient(ctx context.Context) (*Client, error) {
 }
 
 // FIXME: this probably shouldn't be here
```

Comment:
after thinking more on this FIXME, I'm thinking that we should make `RepoTree` (or similar) a more single point of contact for all reads / writes in a repo chaintree. Practically I see injecting tupelo / nodestore in, exposing some finder / create methods on that client, and then returned objects have all the chaintree accessors as well as chaintree modifiers (namely tupelo PlayTransactions)